### PR TITLE
Update apiextensions to v3.4.0

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#13905

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 	github.com/giantswarm/backoff v0.2.0
 	github.com/giantswarm/errors v0.2.3
 	github.com/giantswarm/exporterkit v0.2.0
-	github.com/giantswarm/k8sclient/v4 v4.0.0
+	github.com/giantswarm/k8sclient/v5 v5.0.0
 	github.com/giantswarm/microerror v0.2.1
-	github.com/giantswarm/micrologger v0.3.1
+	github.com/giantswarm/micrologger v0.3.3
 	github.com/giantswarm/operatorkit/v2 v2.0.0
 	github.com/giantswarm/tenantcluster/v3 v3.0.0
```
Comment (tfussell) [inline go.mod:14]: Needs to be updated to v4 I guess

Comment (rossf7) [reply go.mod:14]: Thanks missed that one. Done.

Comment (tfussell) [reply go.mod:14]: I didn't even notice `operatorkit`. I meant `tenantcluster` should be v4 :) Funny coincidence.

Comment (rossf7) [reply go.mod]: Wow so I missed 2!  I think I've looked at too many go.mod files today. 😄 

Should be good this time. PTAL

Comment (tfussell) [reply go.mod]: I know the feeling...

**Review by tfussell — APPROVED** [review]

**Pull request merged**